website/static/js/share/utils.js

```diff
@@ -26,57 +30,124 @@ var formatNumber = function(num) {
 
 var loadingIcon = m('img[src=/static/img/loading.gif]',{style: {margin: 'auto', display: 'block'}});
 
-var loadMore = function(vm) {
-    if (vm.query().length === 0) {
+utils.errorState = function(vm){
+    vm.results = null;
+    vm.statsData = undefined;
+    vm.time = 0;
+    vm.count = 0;
+    vm.resultsLoading(false);
+    m.redraw(true);
+    $osf.growl('Error', 'invalid query');
+};
+
+utils.updateVM = function(vm, data) {
+    if (data === null) {
         return;
     }
-    var page = vm.page++ * 10;
-    vm.resultsLoading(true);
-
-    m.request({
-        method: 'get',
-        background: true,
-        url: '/api/v1/share/?' + $.param({
-            from: page,
-            q: vm.query(),
-            sort: 'dateUpdated',
-        })
-    }).then(function(data) {
-        vm.time = data.time;
-        vm.count = data.count;
-        data.results.forEach(function(result) {
-            result.title = scrubHTML(result.title);
-            result.description = scrubHTML(result.description);
-        });
-
-        vm.results.push.apply(vm.results, data.results);
-
-        vm.resultsLoading(false);
-    }).then(m.redraw).then(function() {
-        callbacks.map(function(cb) {cb();});
+    vm.time = data.time;
+    vm.count = data.count;
+    data.results.forEach(function(result) {
+        result.title = utils.scrubHTML(result.title);
+        result.description = utils.scrubHTML(result.description);
+    });    
+    vm.results.push.apply(vm.results, data.results);   
+    m.redraw();
+    $.map(callbacks, function(cb) {
+        cb();
     });
 };
 
-var search = function(vm) {
-    if (vm.query().length === 0) {
-        return;
+utils.loadMore = function(vm) {    
```

Comment:
which controller? The only place it is used is the results view (which only has access to the results controller) and the search function. The search function would have to be in searchBar.js, where it would not have access to the results controller. Unless I did something like ```ctrl.vm.resultsController.search```, but even then it is not guaranteed to be defined unless the order the controllers are instantiated is correct. Though I do agree that it doesn't fit what would normally be a 'utility' function.